Zend/tests/match/028.phpt

```diff
@@ -0,0 +1,36 @@
+--TEST--
+Test result of match cannot be modified by reference
+--FILE--
+<?php
+
+// opcache can't be certain Test::usesRef is actually this method
+if (!class_exists('Test')) {
+    class Test {
+        public static function usesRef(&$x) {
+            $x = 'modified';
+        }
+        public static function usesValue($x) {
+            echo "usesValue $x\n";
+        }
+    }
+}
+
+function main(int $i): int {
+    Test::usesValue(match(true) { true => $i });
+    Test::usesValue(match($i) { 42 => $i });
+    var_dump($i);
+    Test::usesRef(match(true) { true => $i });
```

Comment:
This also works for references from other sources such as function calls because of ` Tx = QM_ASSIGN Vx` being used consistently

```php
<?php
function &returns_ref() {
    static $var = 0;
    return $var;
}

function accepts_ref(&$var) {
}
function dynamic($x) {
    accepts_ref(match($x) {default => returns_ref(), 1, 2, 3, 4, 5 => 'other'});
}
dynamic(null);
```